src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -844,7 +884,14 @@ private void EmitSequenceExpression(BoundSequence sequence, bool used)
             Debug.Assert(sequence.Value.Kind != BoundKind.TypeExpression || !used);
             if (sequence.Value.Kind != BoundKind.TypeExpression)
             {
-                EmitExpression(sequence.Value, used);
+                if (used && sequence.Type.SpecialType == SpecialType.System_Boolean)
+                {
+                    EmitCondExpr(sequence.Value, sense: sense);
+                }
+                else
+                {
```

Comment:
Can we assert that `sense == true` in this branch?

Reply:
Added an assertion to the same effect.

Reply:
We can get into the alternative branch with `!used` and a boolean type. The added assertion doesn't cover `sense` in that scenario.